File: RecipeAPI/Models/Recipe.cs
using System.Diagnostics.Contracts;

namespace RecipeAPI.Models
{
    public class Recipe
    {
        public string? Id { get; set; }
        public string? Title { get; set; }

        public string? TimeToComplete { get; set; }

        public string? Notes { get; set; }

        public string? ImageData { get; set; }

        public List<Step>? Steps { get; set; }

        public List<Ingredient>? Ingredients { get; set; }
    }
}

File: RecipeSite/src/app/recipe.ts
export interface Recipe {
  id: string;
  title: string;
  timeToComplete: string;
  notes: string;
  imageData: string;
  steps: Step[];
  ingredients: Ingredient[];
}

export interface Ingredient {
  description: string;
  qty: string;
  unit: string;
}

export interface Step {
  num: number;
  instructions: string;
  imageData: string;
}
